npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb

```diff
@@ -458,10 +458,13 @@ def run_npm_checker(path:, version:)
         end
 
         def run_npm7_checker(version:)
-          SharedHelpers.run_shell_command(
+          cmd =
             "npm install #{version_install_arg(version: version)} --package-lock-only --dry-run=true --ignore-scripts"
-          )
-          nil
+          output = SharedHelpers.run_shell_command(cmd)
+          if output.match?(NPM7_PEER_DEP_ERROR_REGEX)
+            error_context = { command: cmd, process_exit_value: 1 }
+            raise SharedHelpers::HelperSubprocessFailed.new(message: output, error_context: error_context)
+          end
```

Comment:
Just to ensure I understand the change, this is an improvement to attach the command being ran to the error context so it's less of a mystery what has happened?

Reply:
In npm 8 some peer dependency conflicts don't result in an error, but they print a warning out instead and return a `0` exit code. In order to keep the change as minimal as possible this detects whenever such a warning is printed and turns it into an error.

I initially looked into handling those warnings separately but the change ended up being much more involved and harder to follow IMO.

Reply:
Ah, I get it, yeah that makes sense. I agree with this approach of keeping our existing behaviour 👍🏻